migration/20170609-1-move-library-configuration-to-configurationsettings.py

```diff
@@ -0,0 +1,152 @@
+#!/usr/bin/env python
+"""Move per-library settings from the Configuration file
+into the database as ConfigurationSettings.
+"""
+import os
+import sys
+import json
+import logging
+from nose.tools import set_trace
+
+bin_dir = os.path.split(__file__)[0]
+package_dir = os.path.join(bin_dir, "..")
+sys.path.append(os.path.abspath(package_dir))
+
+from core.model import (
+    ConfigurationSetting,
+    ExternalIntegration,
+    get_one_or_create,
+    production_session,
+    Library,
+    create,
+)
+
+from api.config import Configuration
+from api.millenium_patron import MilleniumPatronAPI
+from api.sip import SIP2AuthenticationProvider
+from api.authenticator import (
+    BasicAuthenticationProvider,
+    OAuthAuthenticationProvider,
+)
+
+log = logging.getLogger(name="Circulation manager library configuration import")
+
+def log_import(service_name):
+    log.info("Importing configuration for %s" % service_name)
+
+def make_patron_auth_integration(_db, provider):
+    integration, ignore = get_one_or_create(
+        _db, ExternalIntegration, protocol=provider.get('module'),
+        goal=ExternalIntegration.PATRON_AUTH_GOAL
+    )
+
+    # If any of the common Basic Auth-type settings were provided, set them
+    # as ConfigurationSettings on the ExternalIntegration.
+    test_identifier = provider.get('test_username')
+    test_password = provider.get('test_password')
+    if test_identifier:
+        integration.setting(BasicAuthenticationProvider.TEST_IDENTIFIER).value = test_identifier
+    if test_password:
+        integration.setting(BasicAuthenticationProvider.TEST_PASSWORD).value = test_password
+    identifier_re = provider.get('identifier_regular_expression')
+    password_re = provider.get('password_regular_expression')
+    if identifier_re:
+        integration.setting(BasicAuthenticationProvider.IDENTIFIER_REGULAR_EXPRESSION).value = identifier_re
+    if password_re:
+        integration.setting(BasicAuthenticationProvider.PASSWORD_REGULAR_EXPRESSION).value = password_re
+    
+    return integration
+        
+def convert_millenium(_db, integration, provider):
+
+    # Cross-check MilleniumPatronAPI.__init__ to see how these values
+    # are pulled from the ExternalIntegration.
+    integration.url = provider.get('url')
+    auth_mode = provider.get('auth_mode')
+    blacklist = provider.get('authorization_identifier_blacklist')
+    if blacklist:
+        integration.setting(MilleniumPatronAPI.IDENTIFIER_BLACKLIST
+        ).value = json.dumps(blacklist)
+    if auth_mode:
+        integration.setting(MilleniumPatronAPI.AUTHENTICATION_MODE
+        ).value = auth_mode
```

Comment:
That whole thing shouldn't be in the file. I copied the old migration script, modified the end, and didn't change the beginning.

I think it's okay to omit verify_certificate, since it's only supposed to be used on a temporary basis until III fixes your ILS.